frontend/src/components/Lightbox.jsx

```diff
@@ -29,31 +31,66 @@ export const Lightbox = ({ onClose, imageSource }) => {
     };
   }, [onClose]);
 
+  useEffect(() => {
+    if (imageSource) {
+      const controlBar = document.getElementById('control-bar');
+      const heightMatch = imageSource.match(/height="(\d+)"/);
+      const widthMatch = imageSource.match(/width="(\d+)"/);
+      const imgHeight = heightMatch ? heightMatch[1] : 'defaultHeight';
+      const imgWidth = widthMatch ? widthMatch[1] : 'defaultWidth';
+
+      if (imgHeight > window.innerHeight) {
+        let widthScale = window.innerWidth / imgWidth;
+        let heightScale =
+          (window.innerHeight - controlBar.clientHeight - 20) / imgHeight;
+        const scalingFactor = Math.min(heightScale, widthScale);
+
+        let adjustedHeight = imgHeight * scalingFactor;
+        let adjustedWidth = imgWidth * scalingFactor;
+        setAdjustedHeight(adjustedHeight);
+        setAdjustedWidth(adjustedWidth);
+      } else {
+        setAdjustedHeight('100%');
+        setAdjustedWidth('100%');
```

Comment:
Can't we initialize with `'100%'` in `useState()`? That would make the else case obsolete.